Sources/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+CreateEntity.swift

```diff
@@ -0,0 +1,679 @@
+						accountAddress: account.address,
+						resource: stakeClaimResource,
+						mode: .loadAll
+					)).tokens
+					return .init(resource: stakeClaimResourceDetails, tokens: tokenData)
+				}
+
+				return nil
+			}()
+
+			return .init(
+				validator: validatorDetails,
+				stakeUnitResource: stakeUnitResource,
+				stakeClaimTokens: stakeClaimTokens,
+				currentEpoch: currentEpoch
+			)
+		}
+	}
+}
+
+extension OnLedgerEntity.Account.PoolUnitResources {
+	var nonEmpty: OnLedgerEntity.Account.PoolUnitResources {
+		let stakes = radixNetworkStakes.compactMap { stake in
+			let stakeUnitResource: OnLedgerEntity.OwnedFungibleResource? = {
+				guard let stakeUnitResource = stake.stakeUnitResource, stakeUnitResource.amount > 0 else {
+					return nil
+				}
+				return stakeUnitResource
+			}()
+
+			let stakeClaimNFT: OnLedgerEntity.OwnedNonFungibleResource? = {
+				guard let stakeClaimNFT = stake.stakeClaimResource, stakeClaimNFT.nonFungibleIdsCount > 0 else {
+					return nil
+				}
+				return stakeClaimNFT
+			}()
+
+			if stakeUnitResource != nil || stakeClaimNFT != nil {
+				return OnLedgerEntity.Account.RadixNetworkStake(
+					validatorAddress: stake.validatorAddress,
+					stakeUnitResource: stakeUnitResource,
+					stakeClaimResource: stakeClaimNFT
+				)
+			}
+			return nil
+		}
+
+		return .init(radixNetworkStakes: stakes, poolUnits: poolUnits)
+	}
+}
+
+extension OnLedgerEntitiesClient {
+	public struct OwnedStakeDetails: Hashable, Sendable {
+		public let validator: OnLedgerEntity.Validator
+		public let stakeUnitResource: ResourceWithVaultAmount?
+		public let stakeClaimTokens: NonFunbileResourceWithTokens?
+		public let currentEpoch: Epoch
+	}
+
+	public struct OwnedResourcePoolDetails: Hashable, Sendable {
+		public let address: ResourcePoolAddress
+		public let poolUnitResource: ResourceWithVaultAmount
+		public let xrdResource: ResourceWithVaultAmount?
+		public let nonXrdResources: [ResourceWithVaultAmount]
+	}
+
+	public struct ResourceWithVaultAmount: Hashable, Sendable {
+		public let resource: OnLedgerEntity.Resource
+		public let amount: RETDecimal
+	}
+
+	public struct NonFunbileResourceWithTokens: Hashable, Sendable {
+		public let resource: OnLedgerEntity.Resource
+		public let tokens: [OnLedgerEntity.NonFungibleToken]
+	}
+}
+
+extension Array where Element == OnLedgerEntity.OwnedFungibleResource {
+	func sorted() async -> OnLedgerEntity.OwnedFungibleResources {
+		@Dependency(\.gatewaysClient) var gatewaysClient
+
+		var xrdResource: OnLedgerEntity.OwnedFungibleResource?
+		var nonXrdResources: [OnLedgerEntity.OwnedFungibleResource] = []
+
+		let networkId = await gatewaysClient.getCurrentNetworkID()
+
+		for resource in self {
+			if resource.resourceAddress.isXRD(on: networkId) {
+				xrdResource = resource
+			} else {
+				nonXrdResources.append(resource)
+			}
+		}
+
+		let sortedNonXrdResources = nonXrdResources.sorted { lhs, rhs in
```

Comment:
might be useful to let `OnLedgerEntity.OwnedFungibleResource` conform to `Comparable` thus allowing this rather long compare function be reusable?